lib/tpm2_policy.c

```diff
@@ -265,25 +260,17 @@ tool_rc tpm2_policy_build_policysecret(ESYS_CONTEXT *ectx,
         tpm2_session *policy_session, tpm2_loaded_object *auth_entity_obj,
         INT32 expiration, TPMT_TK_AUTH **policy_ticket,
         TPM2B_TIMEOUT **timeout, TPM2B_NONCE *nonce_tpm,
-        const char *policy_qualifier_path) {
+        const char *policy_qualifier_data) {
 
     /*
      * Qualifier data is optional. If not specified default to 0
      */
-    unsigned long file_size = 0;
-    bool result = true;
-    if (policy_qualifier_path) {
-        result = files_get_file_size_path(policy_qualifier_path, &file_size);
-        if (!result) {
-            return tool_rc_general_error;
-        }
-    }
-
-    TPM2B_NONCE policy_qualifier = { .size = (uint16_t) file_size };
```

Comment:
@idesai this code is wrong, and broken. file_size is a long, which can hold a value larger than 16 bits. You could be reading only partial files as you cast it done without checking that it is not larger than UINT16_MAX.

Additionally, this code was over complicated... all you need to do is:
1. set a size to the max size of the buffer
2. call files_load_bytes_from_path(), which will read up to max bytes and error out if the file hasn't been fully consumed as well as update the size parameter to the actual size in bytes read.
3. go on your way using the contents, validate size if your expecting something certain.